openwillis/measures/video/preprocess_video.py

```diff
@@ -554,132 +571,109 @@ def cluster_facedata(
-    n_sd : float
-        Number of standard deviations above which to consider a jump significant.
-    threshold : float
-        Threshold for considering segments to be in the same location based on mean `bb_x` and `bb_y`.
+    bbox_cols : list
+        List of column names for bounding box values.
+    interpolate : bool
+        Whether to interpolate bounding box values in the output.
 
     Returns:
     ............
-    frames_to_keep : list
-        List of frame indices to keep after filtering.
-
+    frame_idx_not_meeting_min : list
+        List of frame indices that do not meet the minimum threshold.
     ---------------------------------------------------------------------------------------------------
-
     """
+    interpolated_section_dfs = []
     cluster_df['cluster_presences'] = np.cumsum(
         cluster_df.sample_time.diff() > ((frames_per_row / fps)*1.5) # 1.5 is hacky but just gives rounding room for fps and frames per row
     )
+    for _, presence_df in cluster_df.groupby('cluster_presences'):
+        n_frames_present = len(presence_df) * frames_per_row
+        if n_frames_present > min_frames_face_present:
 
-    frame_idx_not_meeting_min = get_frame_indices_below_threshold(
-        cluster_df, min_frames_face_present, frames_per_row
-    )
-
-    frames_to_cut_bools = cluster_df.frame_idx.isin(frame_idx_not_meeting_min)
-    cluster_df = cluster_df.loc[~frames_to_cut_bools]
-    
-    return cluster_df
-
-
-def convert_face_df_to_bbox_list(
-    face_df,
-    frames_per_row, 
-    max_frame_idx, 
-    num_frames_vid
-):
-    """
-    ---------------------------------------------------------------------------------------------------
-
-    Prepares the face dataframe for output by interpolating missing frames and converting it to a list of dictionaries.
-
-    Parameters:
-    ............
-        face_df (pandas.DataFrame): The input face dataframe.
-        min_frame_idx (int): The minimum frame index.
-        max_frame_idx (int): The maximum frame index.
-        bbox_cols (list): The list of column names representing bounding box coordinates.
-        interpolate_col (str, optional): The column name to use for interpolation. Defaults to 'frame_idx'.
-
-    Returns:
-    ............
-        list: A list of dictionaries representing the processed face dataframe.
-    ---------------------------------------------------------------------------------------------------
+            section_df = create_section_output(
+                presence_df,
+                bbox_cols,
+                interpolate
+            )
 
-    """
-    max_frame = min(num_frames_vid, max_frame_idx)
-    out_data_list = [{} for _ in range(max_frame)]
-    for idx,row in face_df.iterrows():
-        # upsample frames
-        for i in range(row.frame_idx, row.frame_idx + frames_per_row):
+            interpolated_section_dfs.append(section_df)
+             
+    df_for_all_presences = pd.concat(interpolated_section_dfs)
```

Comment:
```suggestion
    df_for_all_presences = pd.concat(interpolated_section_dfs) if interpolated_section_dfs else pd.DataFrame()
```

Here I'd add a check to avoid error if `interpolated_section_dfs` remains empty